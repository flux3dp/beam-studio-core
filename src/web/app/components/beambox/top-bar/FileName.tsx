import React from 'react';

import i18n from 'helpers/i18n';
import { useIsMobile } from 'helpers/system-helper';

const LANG = i18n.lang.topbar;

interface Props {
  fileName: string;
  hasUnsavedChange: boolean;
}

function FileName({ fileName, hasUnsavedChange }: Props): JSX.Element {
  const isMobile = useIsMobile();
  if (window.os === 'Windows' || isMobile) {
    return null;
  }

  return (
    <div className="file-title">
      {(fileName || LANG.untitled) + (hasUnsavedChange ? '*' : '')}
    </div>
  );
}

export default FileName;
