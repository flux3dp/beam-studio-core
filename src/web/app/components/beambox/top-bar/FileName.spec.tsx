import React from 'react';
import { render } from '@testing-library/react';

import FileName from './FileName';

const mockUseIsMobile = jest.fn();
jest.mock('helpers/system-helper', () => ({
  useIsMobile: (...args) => mockUseIsMobile(...args),
}));

jest.mock('helpers/i18n', () => ({
  lang: {
    topbar: {
      untitled: 'Untitled',
    },
  },
}));

describe('test FileName', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render correctly', () => {
    mockUseIsMobile.mockReturnValue(false);
    const { container } = render(<FileName
      fileName="abc.svg"
      hasUnsavedChange
    />);
    expect(container).toMatchSnapshot();

    const { container: container2 } = render(<FileName
      fileName=""
      hasUnsavedChange={false}
    />);
    expect(container2).toMatchSnapshot();
  });

  it('should hide on windows', () => {
    mockUseIsMobile.mockReturnValue(false);
    window.os = 'Windows';
    const { container } = render(<FileName
      fileName="abc.svg"
      hasUnsavedChange
    />);
    expect(container).toBeEmptyDOMElement();
  });

  it('should hide on mobile', () => {
    mockUseIsMobile.mockReturnValue(true);
    const { container } = render(<FileName
      fileName="abc.svg"
      hasUnsavedChange
    />);
    expect(container).toBeEmptyDOMElement();
  });
});
