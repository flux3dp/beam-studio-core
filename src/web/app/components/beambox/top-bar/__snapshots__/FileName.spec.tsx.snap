// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test FileName should render correctly 1`] = `
<div>
  <div
    class="file-title"
  >
    abc.svg*
  </div>
</div>
`;

exports[`test FileName should render correctly 2`] = `
<div>
  <div
    class="file-title"
  >
    Untitled
  </div>
</div>
`;
